src/lexer/Lexer.ts

```diff
@@ -655,15 +652,10 @@ export class Lexer {
             // replace the exponential marker with a JavaScript-friendly "e"
             asString = this.source.slice(this.start, this.current).replace(/[dD]/, 'e');
             this.addToken(TokenKind.DoubleLiteral);
-            return;
-        }
-
-        if (designator === '!') {
+        } else if (designator === '!') {
```

Comment:
I combined several separate if statements into one continuous if...else if...else if... block. This cleaned up the code by removing several `return` statements and making it easier to follow the flow.